packages/next/src/lib/metadata/metadata.tsx

```diff
@@ -197,17 +196,17 @@ async function getResolvedMetadataImpl(
 ): Promise<React.ReactNode> {
   const errorConvention = errorType === 'redirect' ? undefined : errorType
 
-  const metadataItems = await resolveMetadataItems(
+  const resolvedMetadata = await resolveMetadata(
     tree,
     searchParams,
     errorConvention,
     getDynamicParamFromSegment,
     createServerParamsForMetadata,
-    workStore
-  )
-  const elements: Array<React.ReactNode> = createMetadataElements(
-    await accumulateMetadata(metadataItems, metadataContext)
+    workStore,
+    metadataContext
   )
+  const elements: Array<React.ReactNode> =
+    createMetadataElements(resolvedMetadata)
```

Comment:
I like that we're removing the step of calling `accumulateMetadata` if it's always needed.

Should we do the same refactor for the  `createMetadataElements` step, such that the only public API we export is the function that gives us an array of React nodes? Or is there a reason for them being independent for now?

Reply:
metadata.tsx was already the rennderer part, I just create another render method to reuse them in the metadata.tsx.
resolve-metadata.ts can still keep the simple one purpose responsibily of resolving and no need to touch jsx